runtime/etc/debug/chpl_lldb_pretty_print.py

```diff
@@ -17,10 +17,67 @@
 # limitations under the License.
 #
 
+from functools import cache
 import lldb
 import re
 
 
+class Ptr:
+    def __init__(self, target, ptr, owned=True):
+        self.target = target
+        assert self.target is not None
+        self.ptr = ptr
+        self.owned = owned
+
+    def __del__(self):
+        if self.owned and self.ptr is not None and self.ptr.IsValid():
+            self.target.EvaluateExpression(
+                f"chpl_debug_free((void*){self.ptr.GetValueAsUnsigned()})"
+            )
+
+    def __call__(self):
+        if self.ptr is not None:
+            return self.ptr
+        else:
+            # return an invalid SBValue
+            return lldb.SBValue()
+
+
+def debugp(*args, **kwargs):
+    print("[chpl_lldb_pretty_print]", *args, **kwargs)
+
+
+def errorp(*args, **kwargs):
+    print("[chpl_lldb_pretty_print][ERROR]", *args, **kwargs)
+
+
+def ResolveWidePointer(wideptr, size=None):
+    locale = wideptr.GetNonSyntheticValue().GetChildMemberWithName("locale")
+    addr = wideptr.GetNonSyntheticValue().GetChildMemberWithName("addr")
+    addr_type = addr.GetType()
+
+    target = wideptr.GetTarget()
+
+    if size is None:
+        size = addr_type.GetPointeeType().GetByteSize()
+    ret_val = target.EvaluateExpression(
+        f"chpl_debug_get((void*){locale.GetLoadAddress()}, (void*){addr.GetValueAsUnsigned()}, {size})"
+    )
+    if not ret_val.IsValid():
+        return Ptr(target, None, owned=False)
+    resolved_ptr = ret_val.Cast(addr_type)
+    if not resolved_ptr.IsValid():
+        return Ptr(target, None, owned=False)
+    return Ptr(target, resolved_ptr)
+
+
+def MaybeResolveWidePointer(wideptr, size=None):
+    if wideptr.GetTypeName().startswith("wide("):
+        return ResolveWidePointer(wideptr, size=size)
+    else:
+        return Ptr(wideptr.GetTarget(), wideptr, owned=False)
```

Comment:
calling this `wideptr` here is confusing because in the else branch you return a non-wide pointer but call it `wide`